server/events/working_dir.go

```diff
@@ -290,6 +333,64 @@ func (w *FileWorkspace) forceClone(log logging.Logger,
 	return nil
 }
 
+func (w *FileWorkspace) forceCloneSha(log logging.Logger, cloneDir string, baseRepo models.Repo, sha string) error {
+	logFields := map[string]interface{}{
+		"repository": baseRepo.FullName,
+		"sha":        sha,
+		"workspace":  cloneDir,
+	}
+
+	err := os.RemoveAll(cloneDir)
+	if err != nil {
+		return errors.Wrapf(err, "deleting dir %q before cloning", cloneDir)
+	}
+
+	// Create the directory and parents if necessary.
+	log.Info(fmt.Sprintf("creating dir %q", cloneDir), logFields)
+	if err := os.MkdirAll(cloneDir, 0700); err != nil {
+		return errors.Wrap(err, "creating new workspace")
+	}
+
+	// During testing, we mock some of this out.
+	baseCloneURL := baseRepo.CloneURL
+	if w.TestingOverrideBaseCloneURL != "" {
+		baseCloneURL = w.TestingOverrideBaseCloneURL
+	}
+
+	// clones the default branch into the clone directory, cds to new repo, and checks out sha
+	cmds := [][]string{
+		{
+			"git", "clone", "--branch", baseRepo.DefaultBranch, "--single-branch", baseCloneURL, cloneDir,
+		},
+		{
+			"cd", cloneDir,
+		},
+		{
+			"git", "checkout", sha,
+		},
+	}
+
+	for _, args := range cmds {
+		cmd := exec.Command(args[0], args[1:]...) // nolint: gosec
+		cmd.Dir = cloneDir
+		// The git merge command requires these env vars are set.
+		cmd.Env = append(os.Environ(), []string{
+			"EMAIL=atlantis@runatlantis.io",
+			"GIT_AUTHOR_NAME=atlantis",
+			"GIT_COMMITTER_NAME=atlantis",
+		}...)
+
+		cmdStr := w.sanitizeGitCredentialsBase(strings.Join(cmd.Args, " "), baseRepo)
```

Comment:
This seems really pointless if its just for error messages, let's just leave it out of the error message.  Id rather just not loop through these commands and run them one by one with proper unique error messaging